### PR TITLE
Automatically compile to arm64 when running on apple silicon and add custom user agent option

### PR DESCRIPTION
**Pull request by Arctic26**
---
Automatically compile to arm64 when running on apple silicon and and add custom user agent option
---
**Pull Request Type**
Please select what type of pull request this is:
- [ ] Bugfix
- [x] Feature Implementation

**Description**
- Automatically selects arm64 architecture when building on an apple chip
- Change the default user agent to hide that freetube is being used

**Desktop (please complete the following information):**
 - OS: macOS
 - OS Version: 12.0.1
 - FreeTube version: 0.15.1

### CODE REVIEW

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
U need to change branch to development

File: src/main/index.js [hunk 1]
```diff
@@ -205,6 +205,7 @@ function runApp() {
         commonBrowserWindowOptions
       )
     )
+    newWindow.webContents.userAgent = "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.69 Safari/537.36" // Prevent youtube from seeing the freetube header
```
Comment (PikachuEXE) [inline src/main/index.js:208]: I am not sure if we should set a custom user agent
Even if we do it should be an option (or at least not a constant value for all OS)

I think this should be reverted

**Review by PikachuEXE — CHANGES_REQUESTED** [review]
User agent change should be reverted

File: _scripts/build.js [hunk 2]
```diff
@@ -8,9 +8,17 @@ const args = process.argv
...
 
 if (platform == 'darwin') {
-  targets = Platform.MAC.createTarget()
+  let arch;
+
+  if (cpus[0].model.startsWith('Apple')) {
+    arch = Arch.arm64;
+  } else {
+    arch = Arch.x64;
+  }
```
Comment (PikachuEXE) [inline _scripts/build.js:20]: Remove trailing commas, set default value...
```suggestion
  let arch = Arch.x64

  if (cpus[0].model.startsWith('Apple')) {
    arch = Arch.arm64
  }

  targets = Platform.MAC.createTarget(['dmg'], arch)
```

**Review by PikachuEXE — CHANGES_REQUESTED** [review]

**Review by PikachuEXE — CHANGES_REQUESTED** [review]
I have tested `_scripts/build.js` changes locally and the built binary seems running fine and reported as "Apple"

image

However we would only accept that change in this PR and not the custom user agent options
Please submit those changes in a separate PR

**Pull request closed without merging**